app/interactors/update_delivery_schedules_for_product.rb

```diff
@@ -0,0 +1,16 @@
+class UpdateDeliverySchedulesForProduct
+  include Interactor
+
+  def perform
+    require_in_context :product
+
+    markets = product.organization.all_markets
+
+    product.delivery_schedules = if product.use_all_deliveries?
+      DeliverySchedule.where(market: markets).visible
+    else
+      product.delivery_schedules.where(market: markets)
```

Comment:
@thermistor what does this line do? Is it just ensuring that any deliveries that are for now-unassociated markets are removed? I'll likely need to modify this for LO-1091 to take new delivery schedules as input in the case where `!product.use_all_deliveries?`. Hard to explain. Should discuss. I'll base my branch on yours.